tests/test_target.py

```diff
@@ -14,35 +15,69 @@
 # limitations under the License.
 #
 
+""" Module for testing different targets. """
+
 import os
-import shutil
-import tempfile
 from unittest import TestCase
-
-from devlib import LocalLinuxTarget
-
+from devlib import AndroidTarget, LinuxTarget, LocalLinuxTarget
+from devlib.utils.misc import get_random_string
 
 class TestReadTreeValues(TestCase):
+    """ Class testing Target.read_tree_values_flat() """
+
+    def test_read_multiline_values(self, target):
+        """ Test Target.read_tree_values_flat() """
+        method_name = self.__class__.test_read_multiline_values.__qualname__
 
-    def test_read_multiline_values(self):
         data = {
             'test1': '1',
             'test2': '2\n\n',
             'test3': '3\n\n4\n\n',
         }
 
-        tempdir = tempfile.mkdtemp(prefix='devlib-test-')
```

Comment:
I'd recommend using tempfile.TemporaryDirectory as a context manager for proper resource disposal. It also allows specifying a base path so we can make it create the folder in the target working dir as well if needed

Reply:
Can `tempfile` create files/folder on remote machine via SSH? I thought it cannot and decided to use `Target.makedirs()`.

Reply:
> Can tempfile create files/folder on remote machine via SSH?

It can't but tempfile was being used before so the test was never about remotely creating a folder is it ? Or was it using a local target and you generalized to any kind ? I'll have to look again at the test modification if you re-upload it (I think you dropped the commit or that part of the change).

Also it might be a good idea to introduce a NamedTemporaryFile/TemporaryDirectory equivalent to devlib, it should be a good simple exercise to do and can be done behind a single method, e.g.:
```
class Target:
    ...
    def make_temp(self, type, prefix=None):
         ...
``` 
with `type` being `"file"` or `"folder"`. The mktemp command of busybox can be used to safely create a unique file/folder and `contextlib.contextmanager` can be used to make a context manager.